File: src/content/solutions/1582_special-positions-in-a-binary-matrix/solution.py
class Solution:
    def numSpecial(self, mat: list[list[int]]) -> int:
        r, c = len(mat), len(mat[0])
        rsum, csum = [0]*r, [0]*c

        for i in range(r):
            for j in range(c):
                if mat[i][j] == 1:
                    rsum[i] += 1
                    csum[j] += 1
        
        ans = 0
        for i in range(r):
            for j in range(c):
                if mat[i][j] == 1 and rsum[i] == 1 and csum[j] == 1:
                    ans += 1

        return ans

File: src/content/solutions/1582_special-positions-in-a-binary-matrix/solution.cpp
#include <vector>
using namespace std;

class Solution {
    public:
        int numSpecial(vector<vector<int>>& mat) {
            int r = mat.size(), c = mat[0].size();
            vector<int> rSum(r, 0), cSum(c, 0);

            for (int i = 0; i < r; i++) {
                for (int j = 0; j < c; j++) {
                    if (mat[i][j] == 1) {
                        rSum[i]++;
                        cSum[j]++;
                    }
                }
            }
            int ans = 0;
            for (int i = 0; i < r; i++) {
                for (int j = 0; j < c; j++) {
                    if (mat[i][j] == 1 && rSum[i] == 1 && cSum[j] == 1) {
                        ans++;
                    }
                }
            }
            return ans;
        }
};